### PR TITLE
Add debug option to visualize wide path flags.

### PR DESCRIPTION
**Pull request by Broxzier**
This implements #7694 

dynamite dunes 2018-06-15 13-47-17

### CODE REVIEW

**Comment by deurklink** [comment]
That was quick! The example image also nicely demonstrates that wide paths could probably be optimized to create a more efficient path.

File: src/openrct2-ui/windows/DebugPaint.cpp [hunk 1]
```diff
@@ -27,21 +27,23 @@
...
 {
     WIDX_BACKGROUND,
+    WIDX_TOGGLE_SHOW_WIDE_PATHS,
     WIDX_TOGGLE_SHOW_BLOCKED_TILES,
     WIDX_TOGGLE_SHOW_SEGMENT_HEIGHTS,
     WIDX_TOGGLE_SHOW_BOUND_BOXES,
     WIDX_TOGGLE_SHOW_DIRTY_VISUALS,
 };
 
 #define WINDOW_WIDTH    (200)
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/DebugPaint.cpp]: While you're at it, could you make this box a little wider? It's too narrow to fit most non-English strings.

File: src/openrct2-ui/windows/DebugPaint.cpp [hunk 2]
```diff
@@ -27,21 +27,23 @@
...
+#define WINDOW_HEIGHT   (8 + 15 + 15 + 15 + 15 + 11 + 8)
 
 static rct_widget window_debug_paint_widgets[] = {
-    { WWT_FRAME,    0,  0,  WINDOW_WIDTH - 1,   0,                  WINDOW_HEIGHT - 1,      0xFFFFFFFF,                             STR_NONE },
-    { WWT_CHECKBOX, 1,  8,  WINDOW_WIDTH - 8,   8,                  8 + 11,                 STR_DEBUG_PAINT_SHOW_BLOCKED_TILES,     STR_NONE },
-    { WWT_CHECKBOX, 1,  8,  WINDOW_WIDTH - 8,   8 + 15,             8 + 15 + 11,            STR_DEBUG_PAINT_SHOW_SEGMENT_HEIGHTS,   STR_NONE },
-    { WWT_CHECKBOX, 1,  8,  WINDOW_WIDTH - 8,   8 + 15 + 15,        8 + 15 + 15 + 11,       STR_DEBUG_PAINT_SHOW_BOUND_BOXES,       STR_NONE },
-    { WWT_CHECKBOX, 1,  8,  WINDOW_WIDTH - 8,   8 + 15 + 15 + 15,   8 + 15 + 15 + 15 + 11,  STR_DEBUG_PAINT_SHOW_DIRTY_VISUALS,     STR_NONE },
+    { WWT_FRAME,    0,  0,  WINDOW_WIDTH - 1,   0,              WINDOW_HEIGHT - 1,  STR_NONE,                               STR_NONE },
+    { WWT_CHECKBOX, 1,  8,  WINDOW_WIDTH - 8,   8 + 15 * 0,     8 + 15 * 0 + 11,    STR_DEBUG_PAINT_SHOW_WIDE_PATHS,        STR_NONE }, // TODO: New string ID
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/DebugPaint.cpp]: Looks like you've already fixed the new string for it?

**Comment by AaronVanGeffen** [comment]
I think automatically resizing it based on the string contents is a little overkill, if I'm honest, but … I guess that works. :stuck_out_tongue:

**Comment by Broxzier** [comment]
At least this way youit'll fit for sure. Unless the player decides to change their language constantly, it's only executed once when opening the window, which doesn't increase the lag spike much at all.

**Pull request merged**